src/ui/ui_manager.py

```diff
@@ -602,6 +642,75 @@ def repair_needed(self) -> bool:
             use_grayscale=True)
         return template_match.valid
 
+    def gambling_needed(self) -> bool:
+        return self._gold_full
+    
+    def set__gold_full (self, bool: bool):
+        self._gold_full = bool
+        self._gambling_round = 1
+
+    def gamble(self, item_finder: ItemFinder):
+        gold = True
+        gamble_on = True
+        if self._config.char["num_loot_columns"]%2==0:
+            ignore_columns = self._config.char["num_loot_columns"]-1
+        else:
+            ignore_columns = self._config.char["num_loot_columns"]-2
+        template_match = self._template_finder.search_and_wait(
+                "REFRESH",
+                best_match=True,
```

Comment:
`best_match` is only used if there multiple templates being searched for and you want to find the one with the closest match on screen.

Reply:
Good to know. Set best_match to False